### PR TITLE
Make the load balancer immediately aware of the number of playing players

### PR DESCRIPTION
**Pull request by freyacodes**
This makes the load balancer more responsive to quick increases in player count, as it will not rely on outdated player data.

The current behavior leads to a sawtooth-esque patterns, like seen with this one node:

image

The original V1 client used this: https://github.com/freyacodes/Lavalink-Client/blob/master/src/main/java/lavalink/client/io/LavalinkLoadBalancer.java#L112

I have not tested this code yet.

### CODE REVIEW

**Review by duncte123 — APPROVED** [review]
On first inspection this seems to do what you want it to do, will merge when you have the graphs of the test results.

Only concern I have is when a client gets connected to a node with other clients connected to it, this is not kept into account with this change

**Comment by freyacodes** [comment]
By testing I mean using it in a small environment to see if it is calculating penalties as expected. That graph has been taken from production, and it would be a while before this change lands there.

**Comment by duncte123** [comment]
> By testing I mean using it in a small environment to see if it is calculating penalties as expected. That graph has been taken from production, and it would be a while before this change lands there.

That makes sense, if the test env gives the wanted results it's fine with me

File: src/main/kotlin/dev/arbjerg/lavalink/internal/loadbalancing/Penalties.kt
Comment (duncte123) [inline src/main/kotlin/dev/arbjerg/lavalink/internal/loadbalancing/Penalties.kt:64]: There is more than one bot on my own nodes, this change should also count the players from other bots on the same node. It might need some tweaking like subtracting the cached players from the stats players, but I am unsure if that improves the accuracy

```suggestion
        val cachedPlayers = node.playerCache.count { it.value.track != null && !it.value.paused }
        val statsPlayers = stats.playingPlayers
        val playerPenalty = max(cachedPlayers, statsPlayers)
```

Comment (freyacodes) [reply src/main/kotlin/dev/arbjerg/lavalink/internal/loadbalancing/Penalties.kt:64]: This is a good compromise, however I will refactor this slightly

**Comment by duncte123** [comment]
Am I ok to merge this?

**Comment by freyacodes** [comment]
> Am I ok to merge this?

I believe so? Do you have reason to be in doubt?

**Comment by duncte123** [comment]
> > Am I ok to merge this?
> 
> I believe so? Do you have reason to be in doubt?

Then I will merge it!

**Pull request merged**
